src/smartHandler.ts

```diff
@@ -329,6 +329,9 @@ export class SMARTHandler implements Authorization {
                     this.fhirVersion,
                 ),
             );
+            if (!Number.isNaN(readResponse.total + 1)) {
+                readResponse.total -= readResponse.entry.length - entries.length;
+            }
```

Comment:
The router should always send the total: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/bundle/bundleGenerator.ts#L32

but if it doesn't lets set it to the existing length:

```suggestion
            if (!readResponse.total) {
                readResponse.total = entries.length;
            } else {
                readResponse.total -= readResponse.entry.length - entries.length;
            }
```

Reply:
Implemented this change, but did it a bit differently as directly modifying `readResponse.total` proved to be a bad idea due to the fact that the test runs `authorizeAndFilterReadResponse` twice per filter (one with the FhirServiceBaseUrl and one without), and the request would end up modified from the previous call. Instead, passing it back as part of the spread operator worked.